{{ cookiecutter.formal_name }}/app/src/main/java/org/beeware/android/MainActivity.java

```diff
@@ -262,6 +263,14 @@ protected void onResume() {
         pythonApp.onResume();
         Log.d(TAG, "onResume() complete");
     }
+	
+    protected void onActivityResult(int requestCode, int resultCode, Intent data)
+    {
+        Log.d(TAG, "onActivityResult() start");
+        super.onActivityResult(requestCode, resultCode, data);
+        pythonApp.onActivityResult(requestCode, resultCode, data);
```

Comment:
@freakboy3742 Should we do as `hasattr` check for `pythonApp.onActivityResult` here? I think it's OK to skip that check if the template can expect a version of Toga with the new `onActivityResult` code in https://github.com/beeware/toga/pull/1158 . Wanted to double-check with you.

Reply:
I get the intention, but I think this will be safe without the `hasattr` guard. We're only going to get an activity result if the code starts an activity... and that won't be happening until the code is merged into Toga to start activities in the first place.